woocommerce/woocommerce
# [Enhancement]: Optimise shipping calculations flow when using the Checkout block

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

We're optimizing shipping calculations flow when using the Checkout block. With this PR, we're making the following changes:

-  We are changing the wording `No shipping options available` in the order summary section to `Enter the address to calculate`, and displaying it in 80% of the font size of the `Shipping` label and normal, neither `<strong>` nor `<em>` when no address is provided.
- We are changing the wording `Shipping options will be displayed here after entering your full shipping address` to `Enter a shipping address to view shipping options` with the text vertically and horizontally centered and a grey background when no address is provided.
- We are changing the wording `Add a shipping address to view shipping options` to `Enter a shipping address to view shipping options` with the text vertically and horizontally centered and a grey background when a partial address is provided.
- When a full address is provided, but no shipping options are available for that address, we're changing the wording `There are no shipping options available. Please check your shipping address.` to `No shipping options are available for this address. Please verify the address is correct or try a different address.`.
- When a full address is provided, but no shipping options are available for that address, we're changing the wording `No shipping options available` in the order summary section to `No available delivery option`, and displaying it in 80% of the font size of the `Shipping` label and normal, neither `<strong>` nor `<em>`.





Closes #46951.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

- Go to `/wp-admin/admin.php?page=wc-settings&tab=shipping&section=options`.
- Enable the option `Enable the shipping calculator on the cart page`.
- Go to `/wp-admin/admin.php?page=wc-settings&tab=shipping&section=pickup_location`
- Disable local pickup.
- Go to `/wp-admin/admin.php?page=wc-settings&tab=shipping&section`.
- Remove all shipping methods.
- Add one shipping method to one country only (Example: UAE).
- Open an incognito window.
- Add a product to the cart.
- Go to the Checkout page that is using the Checkout block.

1. Confirm the text in the order summary section is `Enter the address to calculate`.


| Before | After |
|---|---|
|![image](https://github.com/woocommerce/woocommerce/assets/11503784/9816c342-9b23-4061-ba6d-5a3df213ed25)|![image](https://github.com/woocommerce/woocommerce/assets/11503784/c56ba91b-b86c-49fe-b568-8e5bd2f92ad7)|

2. Confirm the text in the shipping rate selector is `Enter a shipping address to view shipping options` vertically and horizontally centered and with a grey background.



| Before | After |
|---|---|
|![image](https://github.com/woocommerce/woocommerce/assets/11503784/d674706f-1a80-41e3-a79c-75c565401bf5)|![image](https://github.com/woocommerce/woocommerce/assets/11503784/5836a44c-1e2e-4274-8e00-ec8701d39671)|




3. Enter the partial shipping address and confirm the text in the shipping rate selector is `Enter a shipping address to view shipping options` vertically and horizontally centered and with a grey background.



| Before | After |
|---|---|
|![image](https://github.com/woocommerce/woocommerce/assets/11503784/d674706f-1a80-41e3-a79c-75c565401bf5)|![image](https://github.com/woocommerce/woocommerce/assets/11503784/5836a44c-1e2e-4274-8e00-ec8701d39671)|




4. Enter the full shipping address for the other country where no shipping method is available (Example: US).
5. Confirm the text in the order summary section is `No available delivery option`. 

| Before | After |
|---|---|
|![image](https://github.com/woocommerce/woocommerce/assets/11503784/54dbaca0-3380-4262-b065-14df5c83b844)|![image](https://github.com/woocommerce/woocommerce/assets/11503784/67a20d88-8401-4586-9a79-ff6bbd52086a)|

6. Confirm there is a notice in the shipping rate selector, and the text of the notice is `No shipping options are available for this address. Please verify the address is correct or try a different address.`.


| Before | After |
|---|---|
|![image](https://github.com/woocommerce/woocommerce/assets/11503784/ebb98d78-7a46-4608-8ce0-16ef1414346e)|![image](https://github.com/woocommerce/woocommerce/assets/11503784/d9b8a85e-e513-46e3-a926-72699006569a)|




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Optimise shipping calculations flow when using the Checkout block
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
